docs/modules/ROOT/pages/guides/v3-migration/index.adoc

```diff
@@ -0,0 +1,188 @@
+[[v3-migration]]
+= 3.0.0 Migration
+This document lists all breaking changes from version 2.x.y to 3.0.0 and how to update.
+
+== How to upgrade
+Simply update `@neo4j/graphql` using npm or your package manager of choice:
+
+[source, bash, indent=0]
+----
+npm update @neo4j/graphql
+----
+
+== Async schema generation
+Schema generation is now asynchronous. Instead of using the property `schema`, now the method `getSchema` will return the schema
+as a _Promise_. This means that creating a server now requires awaiting for that method:
+
+Instead of
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+const server = new ApolloServer({
+    schema: neoSchema.schema,
+});
+----
+
+Now you'll need to do the following:
+
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+neoSchema.getSchema().then((schema) => {
+    const server = new ApolloServer({
+        schema: schema
+    });
+});
+----
+
+== Relationship changes
+This release contains an overhaul of our relationship validations, this will require a few changes to the schema.
+
+=== Many to * relationships
+To improve consistency and validation, **all** "many to *" relationships need to be defined as _required_ in the schema:
+
+[source, graphql, indent=0]
+----
+type Movie {
+    actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN)
+    director: Director @relationship(type: "DIRECTED", direction: IN)
+}
+----
+
+Note that any other notation, such as `[Actor]` or `[Actor!]` will **not** be valid. "One to one" relationships
+such as `Director` remain unchanged.
+
+=== Relationship cardinality
+Runtime checks for relationships number of connections have been rolled in with this release, this means that some
+databases with inconsistent relationships between the schema definition and the actual data may now fail in some queries.
```

Comment:
Why would this have happened? It will be worthwhile giving some background as to why some databases might have ended up in this state.

Reply:
I can think of manual changes to the DB, bugs, or changes in typedefs as the main culprits, anything else we should list there?

Reply:
More like, due to the fact that we never enforced this before, it's likely that through normal use of our library before this version, your database may have ended up in this state.